CppCoreGuidelines.md

```diff
@@ -9785,19 +9785,16 @@ Consider:
 
     void f()
     {
-        X x;
-        X* p1 { new X };              // see also ???
-        unique_ptr<X> p2 { new X };   // unique ownership; see also ???
-        shared_ptr<X> p3 { new X };   // shared ownership; see also ???
-        auto p4 = make_unique<X>();   // unique_ownership, preferable to the explicit use "new"
-        auto p5 = make_shared<X>();   // shared ownership, preferable to the explicit use "new"
+        X* p1 { new X };              // bad, p1 will leak
+        auto p2 = make_unique<X>();   // good, unique ownership
+        auto p3 = make_shared<X>();   // good, shared ownership
     }
 
 This will leak the object used to initialize `p1` (only).
 
 ##### Enforcement
 
-(Simple) Warn if the return value of `new` or a function call with return value of pointer type is assigned to a raw pointer.
+(Simple) Warn if the return value of `new` or a function returning an owning pointer is assigned to a raw pointer.
```

Comment:
Not for this PR, but should a rule be added that a function should not return owning pointers/raw resources?  I think it would indirectly violate a few rules (R.11, R.12) but would it be worth a rule of its own?

Reply:
Isn't it quite common (and acceptable) for a factory method to return an owning pointer?
``` unique_ptr<X> MakeX(); ```

Reply:
Yes, thanks.  I was trying to mention the ```X* MakeX();``` case.  I reworded it, is that clear now?